cmd/rekor-server/app/serve.go

```diff
@@ -77,6 +79,20 @@ var serveCmd = &cobra.Command{
 			slog.Error(fmt.Sprintf("failed to initialize append options: %v", err))
 			os.Exit(1)
 		}
+		// Compute log ID for TransparencyLogEntry, to be used by clients to look up
+		// the correct instance in a trust root. Log ID is equivalent to the non-truncated
+		// hash of the public key and origin per the signed-note C2SP spec.
+		pubKey, err := signer.PublicKey(options.WithContext(ctx))
+		if err != nil {
+			slog.Error(fmt.Sprintf("failed to get public key: %v", err))
+			os.Exit(1)
+		}
+		_, logID, err := note.KeyHash(viper.GetString("hostname"), pubKey)
```

Comment:
I'm not sure we should be using hostname here? Isn't this service potentially hidden behind a load balancer? Is it doubly used to define what the infrastructure level hostname should be and the externally visible hostname when computing the logId?

Reply:
`hostname` is actually just the origin string, it's not used to serve traffic (those flags are `(http|grpc)-(address|port)`. 

If one isn't set, then the OS hostname is used - https://github.com/sigstore/rekor-tiles/blob/b64cdc986066e24d483c26784c71f62b3cc6f760/cmd/rekor-server/app/serve.go#L151-L156. For any public deployment, we'll always set the hostname.

We could update this to be `--origin` in another PR.

Reply:
Ah interesting. Cool.